editor/src/components/canvas/controls/select-mode/flex-gap-control.tsx

```diff
@@ -0,0 +1,285 @@
+    updatedGapValue.renderedValuePx,
+    flexDirection,
+  )
+
+  const onMouseDown = React.useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      startInteraction(e, dispatch, canvasOffset.current, scale)
+    },
+    [canvasOffset, dispatch, scale],
+  )
+
+  return (
+    <CanvasOffsetWrapper>
+      <div data-testid={FlexGapControlTestId} ref={controlRef}>
+        {controlBounds.map(({ bounds, path: p }) => {
+          const path = EP.toString(p)
+          return (
+            <GapControlSegment
+              key={path}
+              hoverStart={controlHoverStart}
+              hoverEnd={controlHoverEnd}
+              handleHoverStart={handleHoverStart}
+              handleHoverEnd={handleHoverEnd}
+              onMouseDown={onMouseDown}
+              indicatorShown={indicatorShown}
+              path={path}
+              bounds={bounds}
+              flexDirection={flexDirection}
+              indicatorColor={indicatorColor}
+              scale={scale}
+              backgroundShown={backgroundShown}
+              isDragging={isDragging}
+              gapValue={updatedGapValue.value}
+            />
+          )
+        })}
+      </div>
+    </CanvasOffsetWrapper>
+  )
+})
+
+interface GapControlSizeConstants {
+  dragBorderWidth: number
+  paddingIndicatorOffset: number
+  hitAreaPadding: number
+  borderWidth: number
+}
+
+const DefaultGapControlSizeConstants: GapControlSizeConstants = {
+  dragBorderWidth: 1,
+  borderWidth: 0.5,
+  paddingIndicatorOffset: 10,
+  hitAreaPadding: 5,
+}
+
+const gapControlSizeConstants = (
+  constants: GapControlSizeConstants,
+  scale: number,
+): GapControlSizeConstants => ({
+  dragBorderWidth: constants.dragBorderWidth / scale,
+  borderWidth: constants.borderWidth / scale,
+  paddingIndicatorOffset: constants.paddingIndicatorOffset / scale,
+  hitAreaPadding: constants.hitAreaPadding / scale,
+})
+
+interface GapControlSegmentProps {
+  hoverStart: React.MouseEventHandler
+  hoverEnd: React.MouseEventHandler
+  handleHoverStart: (_: string) => void
+  handleHoverEnd: () => void
+  onMouseDown: React.MouseEventHandler
+  bounds: CanvasRectangle
+  flexDirection: SimpleFlexDirection
+  gapValue: CSSNumber
+  indicatorShown: string | null
+  path: string
+  indicatorColor: string
+  scale: number
+  isDragging: boolean
+  backgroundShown: boolean
+}
+
+const GapControlSegment = React.memo<GapControlSegmentProps>((props) => {
+  const {
+    hoverStart,
+    hoverEnd,
+    handleHoverEnd,
+    handleHoverStart,
+    onMouseDown,
+    indicatorShown,
+    bounds,
+    isDragging,
+    gapValue,
+    flexDirection,
+    indicatorColor,
+    scale,
+    path,
+    backgroundShown,
+  } = props
+
+  const { dragBorderWidth, hitAreaPadding, paddingIndicatorOffset, borderWidth } =
+    gapControlSizeConstants(DefaultGapControlSizeConstants, scale)
+  const { width, height } = handleDimensions(flexDirection, scale)
+
+  const handleHoverStartI = React.useCallback(
```

Comment:
handleHoverStart*I*? What's the I for?

Reply:
Ah, in that case can you call it `handleHoverStartInner` please?

Reply:
done!